src/shared/calendarspec.c

```diff
@@ -1196,12 +1197,19 @@ static int tm_within_bounds(struct tm *tm, bool utc) {
                 return negative_errno();
 
         /* Did any normalization take place? If so, it was out of bounds before */
```

Comment:
somewhere here it would be good to add a comment, explaining why we reset those fields if we found a difference

Reply:
Fixed.